src/error.rs

```diff
@@ -44,12 +43,27 @@ pub enum Error {
     TimedOut,
     #[error("unsupported: {0}")]
     Unsupported(String),
+    #[error("error while parsing data from the response: {0}")]
+    ParserError(BoxedError),
+    #[error("struct mismatches the database definition; field {field_name} has unexpected type {unexpected_type}; allowed types for {field_name}: {allowed_types}; database columns: {all_columns:?}")]
+    DataTypeMismatch {
+        field_name: String,
+        allowed_types: String,
+        unexpected_type: String,
+        all_columns: String,
+    },
```

Comment:
This error should be removed, cause I just started using panics in case of validation failure. I think we should just panic in case of a struct mismatch because there is no recovery path, and further execution is unsafe.